File: app_pojav_zh/src/main/java/com/movtery/utils/CleanUpCache.java
package com.movtery.utils;

import static net.kdt.pojavlaunch.Tools.runOnUiThread;

import android.content.Context;
import android.widget.Toast;

import net.kdt.pojavlaunch.R;
import net.kdt.pojavlaunch.Tools;

import org.apache.commons.io.FileUtils;

import java.io.File;

public class CleanUpCache {
    private static boolean isCleaning = false;
    public static void start(Context context) {
        if (isCleaning) return;
        isCleaning = true;

        long totalSize = FileUtils.sizeOfDirectory(Tools.DIR_CACHE);
        try {
            File[] files = Tools.DIR_CACHE.listFiles();
            if (files != null) {
                if (files.length == 0) {
                    runOnUiThread(() -> Toast.makeText(context, context.getString(R.string.zh_clear_up_cache_not_found), Toast.LENGTH_SHORT).show());
                    return;
                }
                for (File file : files) {
                    FileUtils.deleteQuietly(file);
                }

                runOnUiThread(() -> Toast.makeText(context, context.getString(R.string.zh_clear_up_cache_clean_up, PojavZHTools.formatFileSize(totalSize)), Toast.LENGTH_SHORT).show());
            }
        } finally {
            isCleaning = false;
        }
    }
}

File: app_pojav_zh/src/main/java/com/movtery/pojavzh/utils/CleanUpCache.kt
package com.movtery.pojavzh.utils

import android.content.Context
import android.widget.Toast
import com.movtery.pojavzh.utils.file.FileTools
import net.kdt.pojavlaunch.PojavApplication
import net.kdt.pojavlaunch.R
import net.kdt.pojavlaunch.Tools
import org.apache.commons.io.FileUtils
import java.io.File

class CleanUpCache {
    companion object {
        private var isCleaning = false

        @JvmStatic
        fun start(context: Context) {
            if (isCleaning) return
            isCleaning = true

            var totalSize: Long = 0
            var fileCount = 0
            try {
                PojavApplication.sExecutorService.execute {
                    val list = PathAndUrlManager.DIR_CACHE.listFiles()?.let {
                        PathAndUrlManager.DIR_APP_CACHE.listFiles()?.let { it1 ->
                            getList(it, it1)
                        }
                    }

                    PathAndUrlManager.FILE_VERSION_LIST.let {
                        val file = File(it)
                        if (file.exists()) list?.add(file)
                    }

                    list?.let{
                        for (file in list) {
                            ++fileCount
                            totalSize += FileUtils.sizeOf(file)
                            FileUtils.deleteQuietly(file)
                        }
                    }

                    Tools.runOnUiThread {
                        if (fileCount != 0) {
                            Toast.makeText(context,
                                context.getString(R.string.clear_up_cache_clean_up, FileTools.formatFileSize(totalSize)),
                                Toast.LENGTH_SHORT
                            ).show()
                        } else {
                            Toast.makeText(context,
                                context.getString(R.string.clear_up_cache_not_found),
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                }
            } finally {
                isCleaning = false
            }
        }

        private fun getList(vararg filesArray: Array<File>): MutableList<File> {
            val filesList: MutableList<File> = ArrayList()
            for (fileArray in filesArray) {
                filesList.addAll(listOf(*fileArray))
            }

            return filesList
        }
    }
}
